client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-details.tsx

```diff
@@ -1,38 +1,47 @@
 import { recordTracksEvent } from '@automattic/calypso-analytics';
-import { getPlan, PLAN_BUSINESS, PLAN_BUSINESS_MONTHLY } from '@automattic/calypso-products';
+import {
+	getPlan,
+	calculateMonthlyPriceForPlan,
+	PLAN_BUSINESS,
+	PLAN_BUSINESS_MONTHLY,
+} from '@automattic/calypso-products';
 import { CloudLogo, Button, PlanPrice } from '@automattic/components';
 import { Title } from '@automattic/onboarding';
 import { Plans2023Tooltip, useManageTooltipToggle } from '@automattic/plans-grid-next';
 import { useI18n } from '@wordpress/react-i18n';
 import clsx from 'clsx';
 import React, { useState, useEffect } from 'react';
 import ButtonGroup from 'calypso/components/button-group';
-import QueryPlans from 'calypso/components/data/query-plans';
+import QuerySitePlans from 'calypso/components/data/query-site-plans';
 import { useSelectedPlanUpgradeMutation } from 'calypso/data/import-flow/use-selected-plan-upgrade';
 import { useSelector } from 'calypso/state';
-import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
-import { getPlanRawPrice } from 'calypso/state/plans/selectors';
+import { getSitePlan } from 'calypso/state/sites/plans/selectors';
 import { UpgradePlanFeatureList } from './upgrade-plan-feature-list';
 import { UpgradePlanHostingDetails } from './upgrade-plan-hosting-details';
 
 interface Props {
+	siteId: number | undefined;
 	children: React.ReactNode;
 }
 
 export const UpgradePlanDetails = ( props: Props ) => {
 	const { __ } = useI18n();
 	const [ activeTooltipId, setActiveTooltipId ] = useManageTooltipToggle();
 	const [ showFeatures, setShowFeatures ] = useState( false );
-
-	const { children } = props;
 	const [ selectedPlan, setSelectedPlan ] = useState<
 		typeof PLAN_BUSINESS | typeof PLAN_BUSINESS_MONTHLY
 	>( PLAN_BUSINESS );
+
+	const { children, siteId } = props;
+
 	const plan = getPlan( selectedPlan );
-	const planId = plan?.getProductId();
+	const planDetails = useSelector( ( state ) =>
+		siteId ? getSitePlan( state, siteId, selectedPlan ) : null
+	);
 
-	const currencyCode = useSelector( getCurrentUserCurrencyCode );
-	const rawPrice = useSelector( ( state ) => getPlanRawPrice( state, planId as number, true ) );
+	const rawPrice = planDetails
+		? calculateMonthlyPriceForPlan( selectedPlan, planDetails?.rawPrice )
+		: undefined;
```

Comment:
We have a bug that stems from this logic as a result of it happening alongside my PR to show the upgrade credits, which is part of why I suggested we pause. But we need the introductory offer details from this API, so I think it's being explicit about what we should, and should not do.

The core issue is that because we're showing the banner due to my work in https://github.com/Automattic/wp-calypso/pull/92094, we need to show the original/standard price in the main block here. It's probably easiest to explain using the screenshot below:

<img width="817" alt="Screenshot 2024-06-27 at 10 59 16" src="https://github.com/Automattic/wp-calypso/assets/3376401/a0981f15-8f6a-4db9-91c7-ea8956165ea1">

This is a site where I have a $40 upgrade credit. I bought a plan a few months ago, but I have a credit towards a plan upgrade. Due to the banner, we should show the standard price of $25/month. However, your calculation is including the available discount, so it's showing $21.67/month (calculated from $260/12). I think we can work around that to simply use the existing `getSitePlanRawPrice()` selector from `calypso/state/plans/selectors` as this selector _intentionally excludes any discount from the plan upgrade credit_. (i.e. it does exactly what we want!)

```suggestion
	const rawPrice = useSelector( ( state ) => getSitePlanRawPrice( state, siteId, selectedPlan ) );
```

I also double-checked to see what the response looks like for a site that's eligible for the introductory discount, and it looks like the discount and introductory offer details don't overlap, and the selector I mentioned above doesn't look at the introductory offer details.